doc/syntax.rst

```diff
@@ -167,12 +166,13 @@ Examples:
 
     bool flag = True
     str name = "program1"
+    par name = {parameter_name}
```

Comment:
Wait, is this how it works? I thought `{parameter_name}` was a simple template variable. So if you run the blackbird script with `{parameter_name}` set to `"hello"`, then the type should be `str`?
```python
str name = {parameter_name}
```

Reply:
Oh, you're right. I sort of thought ahead and probably didn't fully understand the parameter type.

So `par name = {parameter_name}` does *not* work, while `str name = {parameter_name}` currently does work. And so does:

```python
par array name = 
    {parameter_name}
```

So `par` is not really a type like all of the other types. Do we want to be able to write e.g. `float phase = {phase}` (does not currently work) and thus have `phase` become a placeholder for a float? It might in that case make more sense to write arrays as:

```python
float array name = 
    {phase_0}
    {phase_1}
```

or even allow

```python
float array name = 
    {phase_0}
    3.14
    0.33
    {phase_2}
    2.7
```

and skip the "par" type altogether. :thinking: 

Does this make sense?

Reply:
> and skip the "par" type altogether. 🤔

Yep this make sense! As far as I was aware `par` was an internal implementation detail, and not an actual part of the spec.

Note that `{templated_variables}` aren't really 'variables' or 'parameters' in the sense of the word; a blackbird template cannot be executed by itself. Instead, these are just 'placeholders' --- you need to provide their value alongside the template to generate the _actual_ script that will be executed.

Similar in idea to string formatting in Python:
```pycon
>>> template_string = "Hello {template_variable}"
>>> template_string.format(template_variable="world")
"Hello world"
```

At one point I think we were planning to have a different file extension for blackbird templates (something like `.xbt`) to distinguish them from blackbird scripts (`.xbb`).